### PR TITLE
chore: COPY status

### PR DESCRIPTION
**Pull request by BohuTANG**
I hereby agree to the terms of the CLA available at: https://databend.rs/dev/policies/cla/

## Summary

* add status for COPY
* add memory metrics for memory
* refine the code for better readable

- Closes #issue

### CODE REVIEW

File: src/query/service/src/interpreters/interpreter_copy.rs [hunk 1]
```diff
@@ -254,67 +240,120 @@ impl CopyInterpreter {
...
-            .await?;
+        let to_table = ctx.get_table(catalog, database, table).await?;
 
-        let (mut build_res, source_schema, files) = if let Some(query) = &plan.query {
-            let (build_res, source_schema) = self.build_query(query).await?;
-            (
-                build_res,
-                source_schema,
-                plan.stage_table_info
+        let mut build_res;
```
Comment (BohuTANG) [inline src/query/service/src/interpreters/interpreter_copy.rs:250]: I have careful review these changed codes, the logic is no changed, but some stateful test failed with: 
`+ERROR 1105 (HY000) at line 1: Internal. Code: 1001, Text = Cannot resize empty pipe..`
@JackTan25

Comment (BohuTANG) [reply src/query/service/src/interpreters/interpreter_copy.rs:250]: https://github.com/datafuselabs/databend/actions/runs/5529492495/jobs/10087872738?pr=12062#step:4:454

File: src/query/service/src/interpreters/interpreter_copy.rs [hunk 2]
```diff
@@ -254,67 +240,120 @@ impl CopyInterpreter {
...
-                    to_table,
+                CopyPlanType::CopyIntoTablePlanOption(plan.clone()),
+                source_schema,
+                to_table.clone(),
+            )?;
+
+            // Perf
+            {
+                metrics_inc_copy_read_file_cost_milliseconds(start.elapsed().as_millis() as u32);
+                self.set_status(&format!(
```
Comment (youngsofun) [inline src/query/service/src/interpreters/interpreter_copy.rs:312]: why do we need to metric and status for  `build_append_data_pipeline`?
it does not really copy any data, just builds the pipeline.

Comment (BohuTANG) [reply src/query/service/src/interpreters/interpreter_copy.rs:312]: How can i measure the copy data cost?

Comment (youngsofun) [reply src/query/service/src/interpreters/interpreter_copy.rs:312]: we can get time for 
 
1. special steps when bind and get_partitons
1. the whole pipeline execution time
2. inside each processor for each part/block.

I can refine this in another PR.

File: src/query/service/src/interpreters/interpreter_copy.rs [hunk 3]
```diff
@@ -254,67 +240,120 @@ impl CopyInterpreter {
...
+                build_res = PipelineBuildResult::create();
+                if !files.is_empty() {
+                    self.build_read_stage_table_data_pipeline(
+                        &mut build_res.main_pipeline,
+                        plan,
+                        to_table.get_block_thresholds(),
+                        files.clone(),
+                    )
+                    .await?;
+                }
```
Comment (youngsofun) [inline src/query/service/src/interpreters/interpreter_copy.rs:269]:
```suggestion
                } else {
                    return Ok(build_res);
                }
```

File: src/query/service/src/interpreters/interpreter_copy.rs (hunk 1 above)
Comment (youngsofun) [reply src/query/service/src/interpreters/interpreter_copy.rs:250]: https://github.com/datafuselabs/databend/pull/12062/files#r1260972866

File: src/query/service/src/interpreters/interpreter_copy.rs (hunk 3 above)
Comment (BohuTANG) [reply src/query/service/src/interpreters/interpreter_copy.rs:271]: I have set it in line260: `build_res = PipelineBuildResult::create();`

Comment (BohuTANG) [reply src/query/service/src/interpreters/interpreter_copy.rs:271]: I see, thanks.

**Pull request merged**